flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetWriters.java

```diff
@@ -459,8 +460,9 @@ private TimestampNanoDataWriter(ColumnDescriptor desc) {
 
     @Override
     public void write(int repetitionLevel, TimestampData value) {
-      column.writeLong(
-          repetitionLevel, value.getMillisecond() * 1_000_000L + value.getNanoOfMillisecond());
+      // For nanosecond precision timestamps, convert to nanoseconds
+      long nanos = value.getMillisecond() * 1_000_000L + value.getNanoOfMillisecond();
+      column.writeLong(repetitionLevel, nanos);
```

Comment:
Could we use the same pattern than the original write for the millisecond timestamp?